app/Rules/ExistClientRule.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Rules;
+
+use App\Repositories\ClientRepository;
+use Carbon\Carbon;
+use Closure;
+use DateTime;
+use Illuminate\Contracts\Validation\ValidationRule;
+
+use function Laravel\Prompts\error;
+
+class ExistClientRule implements ValidationRule
+{
+    protected ClientRepository $clientRepository;
+
+    public function __construct()
+    {
+        $this->clientRepository = new ClientRepository();
+    }
+
+    /**
+     * Run the validation rule.
+     *
+     * @param  \Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+
+        function calcularIdade($value)
```

Comment:
Função dentro da função validate isso vai funcionar????

Tá errado 

Se quiser usar função cria fora de outra função, fora do validate, depois chama a função dentro do validate, tipo

Validate($date){
$this->isLegalAge($date)